packages/components/src/Chips/InternalChip.tsx

```diff
@@ -0,0 +1,150 @@
+   * <Chip prefix={<ChipIcon name="quote" />} />
+   * ```
+   */
+  readonly prefix?: ReactElement<ChipAvatarProps | ChipIconProps>;
+
+  /**
+   * Adds a component on the right side of the label.
+   */
+  readonly suffix?: ReactElement<ChipIconProps | ChipButtonProps>;
+
+  /**
+   * Callback when the chip itself gets clicked.
+   */
+  onClick?(event: SyntheticEvent): void;
+}
+
+export function InternalChip({
+  label,
+  active = false,
+  disabled = false,
+  invalid = false,
+  prefix,
+  suffix,
+  warnOnLongLabels = false,
+  onClick,
+}: InternalChipProps) {
+  const component = computedProps();
+  assertProps();
+
+  const className = classnames(styles.chip, {
+    [styles.clickable]: component.isClickable,
+    [styles.active]: active,
+    [styles.disabled]: disabled,
+    [styles.invalid]: invalid,
+    [styles.hasPrefix]: prefix,
+    [styles.hasSuffix]: suffix,
+  });
+
+  const Tag = onClick ? "button" : "div";
+
+  return (
+    <Tag
+      className={className}
+      data-testid="chip-wrapper"
+      {...(component.isClickable && {
+        onClick: onClick,
+        disabled: disabled,
+      })}
+    >
+      {renderPrefix()}
+      <Typography size="base">{label}</Typography>
+      {renderSuffix()}
+    </Tag>
+  );
+
+  function computedProps() {
+    return {
+      isPrefixAvatar: prefix?.type === ChipAvatar || false,
+      isPrefixIcon: prefix?.type === ChipIcon || false,
+      isSuffixIcon: suffix?.type === ChipIcon || false,
+      isSuffixButton: suffix?.type === InternalChipButton || false,
+      isClickable: onClick && !disabled,
+    };
+  }
+
+  function renderPrefix() {
+    if (component.isPrefixIcon) {
+      return recolorChipIcon(prefix as ReactElement<ChipIconProps>);
+    }
+    return prefix;
+  }
+
+  function renderSuffix() {
+    if (component.isSuffixIcon) {
+      return recolorChipIcon(suffix as ReactElement<ChipIconProps>);
+    }
+    return suffix;
+  }
+
+  function recolorChipIcon(icon: ReactElement<ChipIconProps>) {
+    let color: IconColorNames | undefined;
+    active && (color = "white");
+    invalid && !disabled && (color = "criticalOnSurface");
+    disabled && !active && (color = "disabled");
+
+    return (
+      <div className={styles.icon}>
+        {React.cloneElement(icon, { ...(color && { color: color }) })}
+      </div>
+    );
+  }
+
+  function assertProps() {
+    useAssert(
+      !!prefix && !(component.isPrefixAvatar || component.isPrefixIcon),
+      `Prefix prop only accepts "<ChipAvatar />" or "<ChipIcon />" component. You have "${prefix?.type}".`,
+    );
+    useAssert(
+      !!suffix && !(component.isSuffixIcon || component.isSuffixButton),
+      `Prefix prop only accepts "<ChipIcon />" component. You have "${suffix?.type}".`,
+    );
+    useAssert(
+      warnOnLongLabels && label.length > 24,
+      `"${label}" label is too long for a Chip; you might be better off using Checkbox or Radio. ${label.length}/24 characters.`,
```

Comment:
oooh truncation eh... maybe!

Reply:
Actually a small thing I just noticed from this screenshot - can we ensure that there's equal vertical spacing between rows of Chips as between Chips in a row?

Reply:
RE: Long strings, that's mostly for the multi/single select chips. the dismissible ones wouldn't have a restriction. tho yeah, we should truncate really long ones.

Do we have data on the longest name in jobber?

Reply:
>  can we ensure that there's equal vertical spacing between rows of Chips as between Chips in a row?

@chris-at-jobber the multi/single select does. The dismissable won't since it's standalone. Not unless we want default margins on it

Reply:
![image](https://user-images.githubusercontent.com/15986172/134986941-7496ebdb-84fe-41c7-bce4-89385dd57a48.png)

Reply:
> Do we have data on the longest name in jobber?

I don't have this information - but do we have any idea on what would be reasonable for a chip? Ex: what can fit well on a mobile browser? etc?

Reply:
Tada! 🎉 

![image](https://user-images.githubusercontent.com/15986172/135004650-e3584092-1fb5-471f-a0cb-afa1e1039f1a.png)

_Full disclaimer: Only works on chips that are rendered truncated. Resizing the window won't remove the tooltip (or add it when a chip gets truncated)_